dvc/repo/__init__.py

```diff
@@ -648,6 +552,8 @@ def close(self):
         self.scm.close()
 
     def _reset(self):
+        self.__dict__.pop("outs_trie", None)
+        self.__dict__.pop("outs_graph", None)
```

Comment:
Both of these will be very handy in https://github.com/iterative/dvc/issues/4847